.github/workflows/pr-lint.yaml

```
@@ -0,0 +1,33 @@
+name: lint pr
+
+on:
+  pull_request_target:
+    types:
+      - opened
+      - edited
+      - synchronize
+
+permissions:
+  pull-requests: read
+
+jobs:
+  lint-pr:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: amannn/action-semantic-pull-request@v5
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          scopes: |
+            Core
+            OpenAI
+            Orchestration
+            DevOps
+          headerPattern: '^(\w+): (?:\[(\w+)\] )?(.+)$'
```

(Minor)

I would like a `-`, e.g. when refering to `[spring-app]`

```suggestion
          headerPattern: '^(\w+): (?:\[([\w-]+)\] )?(.+)$'
```

---

This regex only defines the capture groups. The allowed topics are (currently) hard coded in the list above. We can use regex there as well, but I felt it isn't needed

---

Interesting... Do I understand correctly [according to the readme](https://github.com/amannn/action-semantic-pull-request), the users of this github action are required to define an allowlisted set of `scopes`?

---

I think defining scopes is optional, didn't test it though. But since scopes themselves are already configured to be optional, I figured we can use a few defined values to prevent `Core` `AiCore` `AICore` `Ai-Core` ...

---

Good idea. Though, without having checked that, I doubt this will work as expected(?)
`[Ai-Core]` will just be part of third capture group, leaving second group empty. Maybe.

---

```suggestion
          headerPattern: '^(\w.+): (?:\[(\w.+)\] )?(.+)$'
```

True, fixed it now ✅